client/state/plugins/marketplace/actions.ts

```diff
@@ -40,3 +67,168 @@ export function setIsPluginInstalledDuringPurchase(
+	state: MARKETPLACE_ASYNC_PROCESS_STATUS;
+	reason?: string;
+} {
+	return {
+		type: MARKETPLACE_PLUGIN_INSTALLATION_STATE_CHANGE,
+		state,
+		reason: reason ?? 'NOT_PROVIDED',
+	};
+}
+
+export function tryPluginInstall( selectedSiteId: number, pluginSlugToBeInstalled: string ) {
+	return function ( dispatch: any, getState: () => any ): void {
+		const state = getState();
+		const { pluginInstallationStatus } = getPurchaseFlowState( state );
+
+		const pluginOnSite = getPluginOnSite( state, selectedSiteId, pluginSlugToBeInstalled );
+		const isPluginInstalled = !! pluginOnSite;
+		const wporgPlugin = getWporgPlugin( state, pluginSlugToBeInstalled );
+
+		const pluginStatus = getStatusForPlugin( state, selectedSiteId, pluginSlugToBeInstalled );
+		const isPluginInformationLoaded = getIsPluginInformationLoaded(
+			state,
+			selectedSiteId,
+			pluginSlugToBeInstalled
+		);
+		if ( pluginInstallationStatus === MARKETPLACE_ASYNC_PROCESS_STATUS.UNKNOWN ) {
+			dispatch( fetchSitePlugins( selectedSiteId ) );
+			dispatch( wporgFetchPluginData( pluginSlugToBeInstalled ) );
+			dispatch( pluginInstallationStateChange( MARKETPLACE_ASYNC_PROCESS_STATUS.FETCHING ) );
+		} else if ( isPluginInformationLoaded ) {
+			if ( ! isPluginInstalled && ( ! wporgPlugin || ! wporgPlugin.slug ) ) {
+				// There is no such plugin on wporg so there is something wrong here
+				// For some reason plugin information becomes unresponsive just right after a transfer
+				// currently we fail silently because it seems to resolve automatically
+				if ( config.isEnabled( 'marketplace-test' ) ) {
+					// eslint-disable-next-line no-console
+					console.error(
+						'::MARKETPLACE::ERROR:: The wporg plugin details could not be fetched or the slug does not exist',
+						{ wporgPlugin }
+					);
+				}
+			} else if ( isPluginInstalled || pluginStatus === 'completed' ) {
+				// This means the plugin was successfully installed earlier, most probably during purchase
+				dispatch( pluginInstallationStateChange( MARKETPLACE_ASYNC_PROCESS_STATUS.COMPLETED ) );
+			} else if ( pluginInstallationStatus !== MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS ) {
+				// If not already started, initiate the plugin install
+				dispatch( installPlugin( selectedSiteId, wporgPlugin ) );
+				dispatch( siteTransferWithPluginInstallTriggered() );
+			} else if ( pluginInstallationStatus === MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS ) {
+				switch ( pluginStatus ) {
+					case PLUGIN_INSTALLATION_COMPLETED:
+						dispatch( pluginInstallationStateChange( MARKETPLACE_ASYNC_PROCESS_STATUS.COMPLETED ) );
+						break;
+					case PLUGIN_INSTALLATION_IN_PROGRESS:
+						dispatch(
+							pluginInstallationStateChange( MARKETPLACE_ASYNC_PROCESS_STATUS.IN_PROGRESS )
+						);
+						break;
+					case PLUGIN_INSTALLATION_ERROR:
+						dispatch(
+							pluginInstallationStateChange(
+								MARKETPLACE_ASYNC_PROCESS_STATUS.ERROR,
+								'Error in plugin installation in the backend'
```

Comment:
Are we going to do anything with those errors? If yes, are they intentionally not translated?

Reply:
Nope these won't be shown to the user right now. It's used as sanity for dev tester if something goes wrong.